fs/report.go

```diff
@@ -184,7 +184,7 @@ func eqDirectory(path string, x, y *directory) []failure {
 		for _, name := range sortedKeys(y.items) {
 			if _, ok := x.items[name]; !ok {
 				yEntry := y.items[name]
-				p = append(p, existenceProblem(name, "unexpected %s", yEntry.Type()))
+				p = append(p, matchGlob(name, yEntry.Type(), x.globs)...)
```

Comment:
With the new function description and now that we are matching globs instead of just filenames I think this needs to be more complicated. We would call `matchGlobs` before checking `x.items[anyFile]`, and instead of comparing filenames in x vs y we'll need to build a map of all the items in x, plus all the matches from `matchGlobs` and compare that map to y.